### PR TITLE
Storage crate reorg and cleanup

### LINKED ISSUE #177: Can't build `azure_storage` without default features or combinations of features less than the default
## What happened

I'm trying to test out [this PR](https://github.com/Azure/azure-sdk-for-rust/pull/176), and in the project where we're using `azure_storage`, we had specified it as `default-features = false, features = ["table", "blob"]`.

If I check out either the issue/175/pr branch at 7cc0997 or the master branch at dbc74334 and try to build just the azure_storage crate, the following commands result in compiler errors:

```
$ cargo build --no-default-features
...
error: aborting due to 93 previous errors
```

```
$ cargo build --no-default-features --features blob
...
error: aborting due to 16 previous errors
```

```
$ cargo build --no-default-features --features table
...
error: aborting due to 85 previous errors
```

```
$ cargo build --no-default-features --features account
...
error: aborting due to 93 previous errors
```

```
$ cargo build --no-default-features --features queue
...
error: aborting due to 85 previous errors
```

Some combinations, like `table,blob`, also don't work. I haven't checked all combinations. `table,blob,queue` *does* work, as does using the default feature set of `account,blob,queue,table`

## What I expected

[The current README for `azure_storage`](https://github.com/Azure/azure-sdk-for-rust/blob/dbc74334a241fb38f7fcb6e28af50b58cacd4e93/sdk/storage/README.md) seems to indicate I should be able to turn off default features and only turn on the ones I'm using, which sounds like each feature should be independent and any combination should compile.

I realize the crate is still going through a lot of reorganization, but I'm wondering what the eventual intent is. In other words, is it the code or the documentation that needs to be fixed?

### PR DESCRIPTION
**Pull request by MindFlavor**
This is a seemly big PR but it's mostly code shuffling. With this PR I believe we are very close to release v0.1! 🥇 

## Purpose

1. Renamed the modules in `blob_storage`, `table_storage`, etc... to avoid module inception (see [https://rust-lang.github.io/rust-clippy/master/index.html#module_inception](https://rust-lang.github.io/rust-clippy/master/index.html#module_inception)).
2. Moved the `blob_storage` specific clients from `core` to the appropriate module.
3. Detangled the modules so the features compile independently (fixes #177). Right now by default it compiles everything but you can, say, use `table_storage` without anything else.
4. Removed hardcoded dependency from `hyper`  (to support #37). 🍾 
5. Removed leftover dependencies no longer used.
6. Clippy'd everything. Most changes are really simple (`ok_or_else` etc...). The only "significant" change is the switch from `Into` to `From`. Now the crate is clippy-compliant. 🍾 
7. Removed **a lot** of old code (most of it unnecessary, see below).

## Cons

1. To remove the `hyper` dependency I had to remove the ability to generate signed URIs. We will reenable it in a future PR.

### CODE REVIEW

File: sdk/storage/Cargo.toml [hunk 1]
```diff
@@ -19,40 +19,184 @@ base64 = "0.13"
...
+data_lake_storage = []
+
+
+
+# This section specifies the required features for the examples. 
+
+[[example]]
+name="table_00"
+required-features = ["table_storage"]
+
```
Comment (rylev) [inline sdk/storage/Cargo.toml:59]: Can we remove all the blank spaces?

File: sdk/storage/Cargo.toml [hunk 2]
```diff
@@ -19,40 +19,184 @@ base64 = "0.13"
...
+[[example]]
+name="put_block_blob_00"
+required-features=["blob_storage"]
+
+[[example]]
+name="put_page_blob_00"
+required-features=["blob_storage"]
+
+#[[example]]
```
Comment (rylev) [inline sdk/storage/Cargo.toml:154]: Can we remove this?

File: sdk/storage/examples/list_blobs_01.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 use azure_core::prelude::*;
-use azure_storage::blob::prelude::*;
+use azure_storage::blob_storage::prelude::*;
```
Comment (rylev) [inline sdk/storage/examples/list_blobs_01.rs:2]: What was the reasoning behind adding the `storage` suffix to all the submodules?

File: sdk/storage/src/blob_storage/blob/mod.rs [hunk 4]
```diff
@@ -204,24 +203,26 @@ impl Blob {
...
 
         let cache_control = h
             .get_as_string(header::CACHE_CONTROL)
-            .unwrap_or("".to_owned());
+            .unwrap_or_else(|| "".to_owned());
 
         let content_disposition = h
             .get_as_string(header::CONTENT_DISPOSITION)
-            .unwrap_or("".to_owned());
+            .unwrap_or_else(|| "".to_owned());
```
Comment (rylev) [inline sdk/storage/src/blob_storage/blob/mod.rs:225]: Nit: I think `String::new` is a bit better than `|| "".to_owned()`

**Review by rylev — COMMENTED** [review]
Looks good. I'm just unsure about the renaming of the submodules to have the `_storage` suffix. I'd like to hear the reasoning for that.

**Comment by MindFlavor** [comment]
For example, the current master has a module table inside a module called table. Clippy complains with: 

image

I cannot think of a better solution than appending `storage` but I'm open to suggestions.

File: sdk/storage/Cargo.toml (hunk 1 above)
Comment (MindFlavor) [reply sdk/storage/Cargo.toml:59]: I did it to separate the different example groups. Will do!

File: sdk/storage/examples/list_blobs_01.rs (hunk 3 above)
Comment (MindFlavor) [reply sdk/storage/examples/list_blobs_01.rs:2]: I've explained below, it's for the module inception: there is a blob module inside the blob module. Now we have the blob module inside the blob_storage module (which I think makes sense)

**Comment by rylev** [comment]
@MindFlavor putting the the table model into a module called `model` would also fix this issue. I'm not against what you've done here, just trying to think about which solution is better.

File: sdk/core/src/util.rs [hunk 5]
```diff
@@ -98,8 +98,8 @@ fn format<D: Display>(value: D) -> String {
 pub fn to_str_without_bom(bytes: &bytes::Bytes) -> Result<&str, std::str::Utf8Error> {
     let s = from_utf8(bytes)?;
 
-    Ok(if s.starts_with('\u{FEFF}') {
-        &s[3..]
+    Ok(if let Some(stripped) = s.strip_prefix('\u{FEFF}') {
```
Comment (rylev) [inline sdk/core/src/util.rs:101]: Nit: I think the following is clearer:

```rust
s.strip_prefix('...').unwrap_or(s)
```

**Review by rylev — APPROVED** [review]
😍 awesome!

**Pull request merged**